src/eos-updater-poll-common.c

```diff
@@ -409,7 +409,7 @@ get_refspec_to_upgrade_on (gchar               **refspec_to_upgrade_on,
   if (checkpoint_ref_for_deployment != NULL)
     {
       /* Set outparams from the checkpoint ref instead */
-      if (collection_ref_to_upgrade_on != NULL)
+      if (collection_ref_to_upgrade_on != NULL && booted_collection_ref != NULL)
```

Comment:
If `(collection_ref_to_upgrade_on != NULL && booted_collection_ref == NULL)`, then this will not set the `*collection_ref_to_upgrade_on` out parameter, which the caller may then check. In that case, it should set it to `NULL`. This needs a follow-up PR.